bifrost/app/blog/blogs/test-your-llm-prompts/src.mdx

```diff
@@ -0,0 +1,286 @@
+
+<BottomLine
+  title="💡 What is online or offline evaluation?"
+  description="Online evaluation uses live data and user interactions to capture real-world scenarios, while offline evaluation tests in controlled environments with past or synthetic data for safe, reproducible pre-deployment assessment. "
+/>
+
+#### Deterministic testing
+
+LLM-as-a-judge is useful for scenarios where outputs are predictable and well-defined, such as:
+
+- Classification tasks (e.g., sentiment analysis - positive or negative).
+- Structured outputs (e.g., JSON validation).
+- Constraint checks (e.g., ensuring no profanity or meet specific formatting rules).
+
+#### Tips for success
+
+- Test the evaluation prompt itself: make sure the prompt is clear and accurate.
+- Use few-shot learning: Include good and bad examples to help guide the evaluation.
+- Temperature setting: Set the evaluation LLM’s temperature to 0 to ensure consistent answers from LLMs.
+
+**Bottom Line**
+
+This approach can **<span style={{color: '#0ea5e9'}}>approximate human judgment</span>** while being more scalable and efficient. But keep in mind that LLM-assisted evaluations can inherit biases from training data, and may or may not outperform human evaluations. Creating an effective evaluation prompt is critical for this method to see reliable results.
+
+## Why evaluating prompts is important (and difficult)
+
+To optimize your LLM application, you need a way to **<span style={{color: '#0ea5e9'}}>quantify its performance</span>**. Choosing the right evaluation method is just as important as the evaluation itself because different evaluators are better suited for certain use cases. For example:
+
+An LLM chatbot designed for **technical support** will have evaluation metrics like:
+
+1. Whether the response directly addresses the specific issue raised by the customer.
+2. Whether the response avoids contradictory information or irrelevant "fluff."
+3. Whether the tone is appropriate (e.g., helpful and professional).
+
+An LLM handles **creative tasks** (i.e. stories or marketing copy) will include metrics like:
+
+4. Whether the response is original, interesting and concise.
+5. Whether the response is aligned with brand guidelines or follow the desired tone.
+6. Whether the response adhere to structural requirements (e.g., word count, formatting).
+
+### Choosing the right metrics
+
+At the end of the day, the choice of evaluation metrics depends on your application’s goals. For example, Retrieval-Augmented Generation (RAG) is ideal for assessing how well an LLM incorporates external knowledge, while domain-specific metrics like BLEU, ROUGE, or accuracy are better for fine-tuned models.
```

Comment:
RAG is not an evaluation metric, it's the name for retrieving external knowledge. Faithfulness is a good one

Reply:
https://docs.lastmileai.dev/autoeval/metrics

some good references for different evaluation metrics